.github/workflows/ci.yml

```diff
@@ -58,3 +59,38 @@ jobs:
         run: python -m pip install -r requirements-dev.txt
       - name: Run check-manifest
         run: check-manifest --no-build-isolation
+  run-e2e-tests:
+    name: Run E2E tests
+    needs: build-dists
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        os: [ ubuntu-latest, macOS-latest, windows-latest ]
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Could we wrap the `push` and main branch conditional in parenthesis? I just want to make sure those are always evaluated first and make that clear.